### ISSUE TITLE
Links using labels or :contents: doesn't work

### ISSUE THREAD

**Issue by samuelabreu** [sphinx-contrib/confluencebuilder#443]
Hi, im using sphinxcontrib-confluencebuilder==1.4.0

When i create an index on a file using:
```
.. contents::
   :local:

```

or internal links using refs:

```
.. _my-label:

Click here :ref:`my-label`
```

or even links to sections as:

```
My Section
-------------

Go to `My Section`_
```

links are created, works on html, but on confluence all those links go anywhere, checking the url it changes to:

`https://confluence/display/SPACE/Pagename+with+space#MySection`

Reading https://support.atlassian.com/confluence-cloud/docs/insert-links-and-anchors/ it should be

`https://confluence/display/SPACE/Pagename+with+space#Pagenamewithspace-MySection`

Couldn't find if its a config missing on sphinx or confluence itself, any hint how to solve that?

**Comment by gec37** [comment]
Hi @jdknight 

Thanks for looking into this.

The issue of additional white space when using `.. contents ::` seems to become apparent only if you have a long string for the title of the page and/or the section headers.

The title always seems to be included in the table of contents whether you use the `:local:` directive or not.

Although the links render as expected when using `:ref:` (with the sphinx.ext.autosectionlabel extension enabled), if you click on any of the links, the first time seems to direct you to the correct part of the page, but after that the links appear broken (i.e. do not re-direct you to the correct location on the page when you click on them).

**Comment by jdknight** [comment]
By chance, does the root `toctree` directive use the `:numbered:` option?

I do see an issue with references when this option is set.

**Comment by jdknight** [comment]
An update on this -- if the reported issues are related to using `:numbered:` in a toctree directive, this issue should be resolved with the changes introduced in #545,

If this is not the case, I have unfortunately not been able to reproduce this issue. As mentioned in some of the comments above, attempts have been made to use [title entries with spaces or extremely large names](https://raw.githubusercontent.com/sphinx-contrib/confluencebuilder/70d41cd01dbebd03c9cbb191d72d75619e4925a8/tests/unit-tests/datasets/toctree-numbered/doc.rst). At this time, the inspected output shows expected results.

If anyone has any example documentation which produces this issue (with relevant configuration options), please let me know. I will in turn be able to locally produce it and add some unit tests to help cover this case.

---

A side note (if I may), if a document is of limited size, for example:

```
simple
======

.. contents::
    :local:


section
-------

this is a test
```

And the total page of a Confluence document is viewable in the view box of a browser:

![image](https://user-images.githubusercontent.com/1834509/141709849-ed8d2ac7-e9ff-44a2-aa77-04b741d2fb85.png)

Clicking on a content link may not appear to have any visual effect. This is due to the page's view already maximized, so there there is no way the page can scroll down to the target. Unfortunately, there is also no way to style the Confluence instance to highlight the anchor target on selection (as theming options are limited to the configured Confluence theme can not controllable by this extension).

This may not be related to what anyone is seeing -- however, mentioning it just in case.

**Comment by jdknight** [comment]
With the details outlined in the above comment and duration since additional feedback has been provided, it is assumed that the specific issue has been resolved. Will mark this issue as closed. If this issue is believed to still exist, please feel free to include additional information of the issue and we can re-open.
